src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -642,6 +662,61 @@ ExecuteSingleSelectTask(CitusScanState *scanState, Task *task)
 
 
 /*
+ * BuildPlacementSelectList builds a list of SELECT placement accesses
+ * which can be used to call StartPlacementListConnection or
+ * GetPlacementListConnection.
+ */
+List *
+BuildPlacementSelectList(char *nodeName, int nodePort, List *relationShardList)
+{
+	ListCell *relationShardCell = NULL;
+	List *placementAccessList = NIL;
+
+	foreach(relationShardCell, relationShardList)
+	{
+		RelationShard *relationShard = (RelationShard *) lfirst(relationShardCell);
+		List *shardPlacementList = NIL;
+		ShardPlacement *placement = NULL;
+		ShardPlacementAccess *placementAccess = NULL;
+		bool missingOK = false;
+
+		shardPlacementList = FinalizedShardPlacementList(relationShard->shardId);
+
+		placement = SearchShardPlacementInList(shardPlacementList, nodeName, nodePort,
+											   missingOK);
+		if (placement == NULL)
+		{
+			ereport(ERROR, (errmsg("no active placement of shard %ld found on node "
+								   "%s:%d",
+								   relationShard->shardId, nodeName, nodePort)));
+		}
+
+		placementAccess = CreatePlacementAccess(placement, PLACEMENT_ACCESS_SELECT);
+		placementAccessList = lappend(placementAccessList, placementAccess);
+	}
+
+	return placementAccessList;
+}
+
+
+/*
+ * CreatePlacementAccess returns a new ShardPlacementAccess for the given placement
+ * and access type.
+ */
+static ShardPlacementAccess *
+CreatePlacementAccess(ShardPlacement *placement, ShardPlacementAccessType accessType)
+{
+	ShardPlacementAccess *placementAccess = NULL;
+
+	placementAccess = (ShardPlacementAccess *) palloc0(sizeof(ShardPlacementAccess));
+	placementAccess->placement = placement;
+	placementAccess->accessType = PLACEMENT_ACCESS_SELECT;
```

Comment:
The function has a accessType parameter, which seems to be ignored in favor of  PLACEMENT_ACCESS_SELECT?